### PR TITLE
Look for built app from dotnet publish in /layers dir

### PR DESCRIPTION
**Pull request by sophiewigmore**
Signed-off-by: Frankie Gallina-Jones <frankieg@vmware.com>

Thanks for contributing. To speed up the process of reviewing your pull request
please provide us with:

* A short explanation of the proposed change:

In order to fix https://github.com/paketo-buildpacks/dotnet-publish/issues/87 we have to make changes to the `dotnet-publish` buildpack that change where the built app executable / runtimeconfig.json files get published. They will now get published to `layers/paketo-buildpacks_dotnet-publish/publish-output`.

In order to get that functionality to work, the `dotnet-execute` buildpack must know to look inside of this layer directory if it exists. This PR is to add that logic. It needs to be merged before the newly added integration tests in `dotnet-publish` can work, since they use this buildpack.

* An explanation of the use cases your change enables:

Please confirm the following:
* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have added an integration test, if necessary.

### CODE REVIEW

File: build.go [hunk 1]
```diff
@@ -13,17 +14,26 @@ func Build(logger scribe.Logger) packit.BuildFunc {
...
 		configParser := NewRuntimeConfigParser()
-		config, err := configParser.Parse(filepath.Join(context.WorkingDir, "*.runtimeconfig.json"))
+		var command string
+		builtAppPath := context.WorkingDir
+
+		_, publishOutputLocationSet := os.LookupEnv("PUBLISH_OUTPUT_LOCATION")
+
+		if publishOutputLocationSet {
+			builtAppPath, _ = os.LookupEnv("PUBLISH_OUTPUT_LOCATION")
+		}
```
Comment (ForestEckhardt) [inline build.go:24]: I think that this block is a bit cumbersome and might flow a bit nicer if you were to just do one os.LookupEnv call.

File: build.go [hunk 2]
```diff
@@ -13,17 +14,26 @@ func Build(logger scribe.Logger) packit.BuildFunc {
 		logger.Title("%s %s", context.BuildpackInfo.Name, context.BuildpackInfo.Version)
 
 		configParser := NewRuntimeConfigParser()
-		config, err := configParser.Parse(filepath.Join(context.WorkingDir, "*.runtimeconfig.json"))
+		var command string
```
Comment (ForestEckhardt) [inline build.go:17]: Does this need to be declared as a variable explicitly? If it does could me move it closer to where it is initially set?

Comment (sophiewigmore) [reply build.go:17]: oops that's left over from a previous implementation of this code. Good catch

**Review by ForestEckhardt — APPROVED** [review]
Looks great!

**Pull request merged**